### PR TITLE
Adapt to the "island" use-case

### PR DESCRIPTION
**Pull request by simlmx**
Generalize the code to support the a single PV site whose capacity changes over time. This is the first part of supporting the "island" use-case.

_See descriptions in individual commits_

### CODE REVIEW

**Review by zakwatts — COMMENTED** [review]
Looks good

File: psp/exp_configs/recent_history.py [hunk 1]
```diff
@@ -13,34 +16,164 @@
 PV_DATA_PATH = "data/5min.netcdf"
 NWP_DATA_PATH = "gs://solar-pv-nowcasting-data/NWP/UK_Met_Office/UKV_intermediate_version_7.zarr"
 
-# TODO define
-ExpConfigBase = object
+# A list of SS_ID that don't contain enough data.
+# I just didn't want to calculate them everytime.
+# TODO Get rid of those when we prepare the dataset.
+SKIP_SS_IDS = [
```
Comment (zakwatts) [inline psp/exp_configs/recent_history.py:22]: Have these just been moved from the dataset.py file? And I'm assuming they are only relevant for a specific project?

Comment (simlmx) [reply psp/exp_configs/recent_history.py:22]: Exactly I moved them to the config.

File: psp/exp_configs/island.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+from psp.data.data_sources.pv import NetcdfPvDataSource, PvDataSource
+from psp.dataset import Splits, split_train_test
+from psp.exp_configs.base import ExpConfigBase
+from psp.models.base import PvSiteModel, PvSiteModelConfig
+from psp.models.recent_history import RecentHistoryModel
+from psp.models.regressors.decision_trees import SklearnRegressor
+from psp.typings import Horizons
+
+PV_DATA_PATH = "data/island/data_15min_FIN.nc"
+NWP_DATA_PATH = "data/island/nwp-v1"
```
Comment (zakwatts) [inline psp/exp_configs/island.py:12]: now "island_nwp_5y"

Comment (simlmx) [reply psp/exp_configs/island.py:12]: Cool! I'll remove it actually as it is not used in here. You can add it in the PR adding NWP.

File: psp/exp_configs/recent_history.py [hunk 3]
```diff
@@ -73,3 +206,16 @@ def get_model(self) -> PvSiteModel:
...
             use_inferred_meta=False,
         )
+
+    def make_dataset_splits(self, pv_data_source: PvDataSource) -> Splits:
+        return split_train_test(
+            pv_data_source,
+            # Starting in 2020 because we only have NWP data from 2020.
+            # TODO Get the NWP data for 2018 and 2019.
+            # train_start = datetime(2018, 1, 1)
+            train_start=dt.datetime(2020, 1, 1),
```
[the remark this answers is not in the document]
Comment (simlmx) [reply psp/exp_configs/recent_history.py:216]: You mean this file? Yes that's the config for the sites that are currently in production.
Actually I should probably change the name of the config and name it according to the name of the "problem" we are solving (like the name of the client), not the name of the underlying model!

**Review by zakwatts — COMMENTED** [review]
All looks good

File: psp/exp_configs/recent_history.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (simlmx) [reply psp/exp_configs/recent_history.py:216]: Yeah "RecentHistory" is mostly the name of the model (which is not a great name but it's harder to change).

The config then defines what model, what dataset, and all the options to use for training, so it's probably a bad idea to name it after the model only!

I'm not sure what you mean by test_config1 though! That's a config used in unit tests, which also uses the RecentHistory model.

**Pull request merged**